build-tools/src/commands/build_command.rs

```diff
@@ -0,0 +1,127 @@
+use std::env;
+use std::io::Result;
+use std::process::Command;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub struct BuildOptions {
+    #[structopt(long, default_value = "amd64")]
+    architecture: String,
+    #[structopt(long, default_value = "123")]
+    agent_version: String,
+    #[structopt(long, default_value = "123")]
+    version: String,
+    #[structopt(long)]
+    context_path: String,
+    #[structopt(long)]
+    destination_path: String,
+    #[structopt(long)]
+    cloudrun: bool,
+}
+
+pub fn build(args: &BuildOptions) -> Result<()> {
+    validate_args(args);
+    match args.cloudrun {
+        true => build_cloud_run(),
+        false => build_extension("cmd/serverless", args),
+    }
+}
+
+fn validate_args(args: &BuildOptions) {
+    validate_architecture_from_string(args.architecture.as_str());
+}
+
+fn validate_architecture_from_string(arch: &str) -> bool {
+    match arch {
+        "arm64" => true,
```

Comment:
I think you can express this as an enum with Struct Opts instead, and that way there will be a better command line error message.

Reply:
yeah I did not about that, this is way better now 

<img width="519" alt="Screen Shot 2022-11-11 at 3 54 03 PM" src="https://user-images.githubusercontent.com/864493/201429519-12d40d4b-9e6d-43d7-af83-2d1be4e8bfdf.png">